apps/aecontract/test/aecontract_SUITE.erl

```diff
@@ -138,7 +139,8 @@ groups() ->
                                  sophia_chain,
                                  sophia_savecoinbase,
                                  sophia_fundme,
-                                 sophia_aens ]}
+                                 sophia_aens,
+                                 sophia_state_handling]}
```

Comment:
Potential additional test cases:
* Test that remote call passing larger state to function returning constant return value (e.g. empty tuple - for removing one variable to test case) increases gas cost. The idea is preventing malicious user to obtain somehow large contract state (potentially paid by other users for legitimate use) then send cheap call tx calling function performing remote call passing large state without paying for cost of passing large state. This would actually be a group of tests - depending on how the state is "larger":
  * State as record with map with more-or-less keys. This is for testing against ["the size of the map does not contribute to the gas costs of a contract call reading or updating it"](https://github.com/aeternity/protocol/blob/2d3246e9bed457f4399e4dde1b3e29ee3af11914/contracts/sophia.md#map-implementation).
  * State as record with map with a key with larger-or-smaller value.
  * State as record with more fields.
  * State as record with one larger-or-smaller non-map field.
* Test that remote call passing constant state and returning larger updated state increases gas cost. The idea is similar to above. Sample cases:
  * State passed is record with a field that is a map. The called contract updates that field with a larger-or-smaller map in its own state (idea is that that may be implemented as assignment of a reference hence very cheap) and returns such updated (larger-or-smaller) state.

These ^^^ are related to gas cost - mainly aimed at clarifying [this comment re gas cost](https://www.pivotaltracker.com/story/show/160259586/comments/194951641), and marginally this item in the description of the ticket:
> Test that you can pass your state as an argument to another contract. (This is possibly very expensive - good implementation might be needed.)

Not sure if ["To be determined: gas costs of `CALL` based on size of calldata and return value"](https://github.com/aeternity/protocol/blame/2d3246e9bed457f4399e4dde1b3e29ee3af11914/contracts/sophia.md#L1059) is tracked anyware as a ticket.

Reply:
The first bullet point is straightforward to test for, I prepared a couple of tests in `contracts_SUITE:sophia_state_gas/1` but disabled the _asserts_ since this does not seem to be implemented yet?! So I think we need to make a ticket and implement this (unless already covered by an existing ticket? @happi ? 

Second part is harder, since the code for making an update to the passed state before returning would also cost gas? But it should be possible to code up something that can be used to compare.